networkdb/networkdb.go

```diff
@@ -427,27 +423,21 @@ func (nDB *NetworkDB) GetTableByNetwork(tname, nid string) map[string]*TableElem
 // table, key) tuple and if the NetworkDB is part of the cluster
 // propagates this event to the cluster.
 func (nDB *NetworkDB) DeleteEntry(tname, nid, key string) error {
-	value, err := nDB.GetEntry(tname, nid, key)
-	if err != nil {
-		return fmt.Errorf("cannot delete entry as the entry in table %s with network id %s and key %s does not exist", tname, nid, key)
+	entry, err := nDB.getEntry(tname, nid, key)
+	if err != nil || entry.deleting {
+		return fmt.Errorf("cannot delete entry %s with network id %s and key %s "+
+			"does not exist or is already being deleted", tname, nid, key)
 	}
 
-	entry := &entry{
-		ltime:    nDB.tableClock.Increment(),
-		node:     nDB.config.NodeID,
-		value:    value,
-		deleting: true,
-		reapTime: nDB.config.reapEntryInterval,
-	}
+	entry.ltime = nDB.tableClock.Increment()
+	entry.node = nDB.config.NodeID
```

Comment:
Is updating `entry.node` necessary in the delete path?

Reply:
yes, because the deletion has the same meaning of a write, so if node A deletes a key of node B, it become the new owner of the key. (Consider that this is all theoretical because current use of it done by libnetwork has a single writer per key)